api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -760,5 +762,16 @@ public User setUserActivationKey(User user) {
 		return user;
 	}
 
+	/**
+	 * @see org.openmrs.api.UserService#changeUserPasswordUsingActivationKey(String, String);
+	 */
+	@Override
+	public void changeUserPasswordUsingActivationKey(String activationKey, String newPassword) throws APIException {
+		User user = getUserByActivationKey(activationKey);
+		if (user == null) {
+			throw new APIException("activation.key.not.correct", (Object[]) null);
```

Comment:
I think you should add a new exception class to throw that way the calling code knows what the issue is and take the necessary steps next. Take a look at other related password and authentication exceptions for an example.